Mathlib/Analysis/InnerProductSpace/JointEigenspace.lean

```diff
@@ -63,50 +65,127 @@ theorem orthogonalFamily_eigenspace_inf_eigenspace (hA : A.IsSymmetric) (hB : B.
     · exact hB.orthogonalFamily_eigenspaces.pairwise h₁ hv2 w hw2
     · exact hA.orthogonalFamily_eigenspaces.pairwise h₂ hv1 w hw1
 
-open Submodule in
-
-/-- The intersection of eigenspaces of commuting selfadjoint operators is equal to the eigenspace of
-one operator restricted to the eigenspace of the other, which is an invariant subspace because the
-operators commute. -/
-theorem eigenspace_inf_eigenspace
-    (hAB : A ∘ₗ B = B ∘ₗ A) (γ : 𝕜) :
-    eigenspace A α ⊓ eigenspace B γ = map (Submodule.subtype (eigenspace A α))
-      (eigenspace (B.restrict (eigenspace_invariant_of_commute hAB α)) γ) :=
-  (eigenspace A α).inf_genEigenspace _ _ (k := 1)
+/-- The joint eigenspaces of a tuple of commuting symmetric operators form an
+`OrthogonalFamily`. -/
+theorem orthogonalFamily_iInf_eigenspaces
+    (hT : ∀ i, (T i).IsSymmetric) :
+    OrthogonalFamily 𝕜 (fun γ : n → 𝕜 ↦ (⨅ j, eigenspace (T j) (γ j) : Submodule 𝕜 E))
+      fun γ : n → 𝕜 ↦ (⨅ j, eigenspace (T j) (γ j)).subtypeₗᵢ := by
+  intro f g hfg Ef Eg
+  obtain ⟨a , ha⟩ := Function.ne_iff.mp hfg
+  have H := (orthogonalFamily_eigenspaces (hT a) ha)
+  simp only [Submodule.coe_subtypeₗᵢ, Submodule.coe_subtype, Subtype.forall] at H
+  apply H
+  · exact (Submodule.mem_iInf <| fun _ ↦ eigenspace (T _) (f _)).mp Ef.2 _
+  · exact (Submodule.mem_iInf <| fun _ ↦ eigenspace (T _) (g _)).mp Eg.2 _
 
 variable [FiniteDimensional 𝕜 E]
 
 /-- If A and B are commuting symmetric operators on a finite dimensional inner product space
-then the eigenspaces of the restriction of B to any eigenspace of A exhaust that eigenspace.-/
-theorem iSup_eigenspace_inf_eigenspace (hB : B.IsSymmetric)
-    (hAB : A ∘ₗ B = B ∘ₗ A):
+then the eigenspaces of the restriction of B to any eigenspace of A exhaust that eigenspace. -/
+theorem iSup_eigenspace_inf_eigenspace (hB : B.IsSymmetric) (hAB : Commute A B) :
     (⨆ γ, eigenspace A α ⊓ eigenspace B γ) = eigenspace A α := by
   conv_rhs => rw [← (eigenspace A α).map_subtype_top]
-  simp only [eigenspace_inf_eigenspace hAB, ← Submodule.map_iSup]
+  have H : ∀ μ, genEigenspace B μ 1 = eigenspace B μ := fun μ ↦ rfl
```

Comment:
You shouldn't write `rfl` lemmas locally in a proof like this. If you need them, it generally means that we're missing `rw` lemmas, but in this case, I think we're not, you just need to use [Module.End.genEigenspace_def](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Eigenspace/Basic.html#Module.End.genEigenspace_def) and [Module.End.eigenspace_def](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Eigenspace/Basic.html#Module.End.eigenspace_def) and `pow_one` (you could put these together into a single lemma if you want)

Reply:
LOL! Indeed I did this because `genEigenspace_one` no longer exists. Let me use those two.

Reply:
Have I committed defeq abuse by doing this? (Just curious.)